src/Auth0.php

```diff
@@ -317,35 +304,19 @@ public function __construct(array $config)
             $this->dontPersist('id_token');
         }
 
-        $session_base_name = ! empty( $config['session_base_name'] ) ? $config['session_base_name'] : SessionStore::BASE_NAME;
-
-        if (isset($config['store'])) {
-            if ($config['store'] === false) {
-                $emptyStore = new EmptyStore();
-                $this->setStore($emptyStore);
-            } else {
-                $this->setStore($config['store']);
-            }
-        } else {
-            $sessionStore = new SessionStore($session_base_name);
-            $this->setStore($sessionStore);
+        $this->store = $config['store'] ?? null;
+        if ($this->store === false) {
+            $this->store = new EmptyStore();
+        } else if (! $this->store instanceof StoreInterface) {
+            $this->store = new SessionStore();
         }
 
-        $this->authStore = $config['auth_store'] ?? null;
-        if (! $this->authStore instanceof StoreInterface) {
-            $this->authStore = new CookieStore();
+        $transientStore = $config['transient_store'] ?? null;
+        if (! $transientStore instanceof StoreInterface) {
+            $transientStore = new CookieStore();
```

Comment:
Name changed from #395

Reply:
Just curious - why are we falling back to an `EmptyStore` above, but not for this transient store?

Reply:
`store` is for identity artifacts (user identity, tokens) and `transientStore` is for required auth transients like nonce and state. Things work fine but are not stored if `store` is empty but things break if `transientStore` is. 

That said, I'll take another look here along with the above.